overlord/snapstate/handlers.go

```diff
@@ -977,15 +977,20 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+func snapdSnapInstalled(st *state.State) bool {
+	var snapst SnapState
+	err := Get(st, "snapd", &snapst)
+	return err == nil
```

Comment:
Maybe this should be `err == nil && snapst.IsInstalled()` then.

Reply:
it's an invariant that there is no SnapState at all if the snap is not installed,   we can argue about Active or not,  though I suspect disabling snapd once it's there is a bad idea (but historically we don't prohibit it)

Reply:
I added the check for `.Active` to be on the safe side but I agree that disabling snapd will probably cause more havoc than just this.